tests/shared/test_arguments.py

```diff
@@ -0,0 +1,76 @@
+import sys
+from argparse import ArgumentParser
+
+import pytest
+
+from pubtools._pulp.arguments import SplitAndExtend
+
+FAKE_OPTION_VALUES = ["value1", "value2", "value3"]
+
+
+@pytest.fixture
+def parser():
+    return ArgumentParser()
+
+
+def test_split_and_extend_single_delimited_instance():
+    """Test a single option instance, with all values in a delimited list."""
+    # test one instance, all delimited using different delimiters
+    for delimiter in ",.-/":
+        parser = ArgumentParser()
+        parser.add_argument(
+            "--option", type=str, action=SplitAndExtend, split_on=delimiter
+        )
+
+        sys.argv = ["fake-command", "--option", delimiter.join(FAKE_OPTION_VALUES)]
+
+        args = parser.parse_args()
+        assert args.option == FAKE_OPTION_VALUES
+
+
+def test_split_and_extend_multiple_option_instances(parser):
+    """Test multiple option instances."""
+    parser.add_argument("--option", type=str, action=SplitAndExtend)
+
+    # test multiple, individual instances
+    sys.argv = ["command"]
+    for value in FAKE_OPTION_VALUES:
+        sys.argv.extend(["--option", value])
+
+    args = parser.parse_args()
+    assert args.option == FAKE_OPTION_VALUES
+
+
+def test_split_and_extend_multiple_mix_and_match_instance():
+    """Test multiple option instances, some with delimited lists."""
+    # test mix-and-match, delimited-and-not, using different delimiters
+    for delimiter in ",.-/":
+        parser = ArgumentParser()
+        parser.add_argument(
+            "--option", type=str, action=SplitAndExtend, split_on=delimiter
+        )
+
+        sys.argv = [
+            "fake-command",
+            "--option",
+            delimiter.join(FAKE_OPTION_VALUES[:-1]),
+            "--option",
+            FAKE_OPTION_VALUES[-1],
+        ]
+
+        args = parser.parse_args()
+        assert args.option == FAKE_OPTION_VALUES
+
+
+def test_split_and_extend_multiple_instances_with_trailing_delimiters(parser):
```

Comment:
The filtering of empty arguments seems like it'll affect more than just trailing delimiters, so it'd be nice to see tests for empty values appearing elsewhere as well, e.g. ",a,b"  , "a,,b" .

Reply:
I went ahead and updated this test to use a variety of cases that would produce empty values in different places and quantities. I am happy to make separate test functions for each instead.

Reply:
I went ahead and updated this test to use a variety of cases that would produce empty values in different places and quantities. I am happy to make separate test functions for each instead.

Reply:
Honestly, if it were me I'd go the opposite direction and have more of these tests combined together and parametrized, such as:

```python
@pytest.mark.parametrize("argv,expected_option", [
  (["--option", "x"], ["x"]),
  (["--option", "x,y"], ["x", "y"]),
  (["--option", "x", "--option", "y"], ["x", "y"]),
  (["--option", "x", "--option", "y,z"], ["x", "y", "z"]),
  (["--option", "x,y", "--option", "a,b"], ["x", "y", "a", "b"]),
  (["--option", "a,,b", "--option", "c,"], ["a", "", "", "b", "c", ""]),
])
def test_split_extend(argv, expected_option, parser, delimiter):
    parser.add_argument("--option", type=str, action=SplitAndExtend, split_on=delimiter)
    sys.argv = ["command"] + argv
    args = parser.parse_args()
    assert args.option == expected_option
```

i.e. there's just a mapping between input values and expected output values, and the test implementation is the same for all cases (you might only need different test functions if you have e.g. some crashing cases or something else where it doesn't work normally).

I find this easier to read, more maintainable and I think it can achieve greater coverage with less test code needed.  This is all quite subjective though and I'm just happy if the relevant cases are tested at all, regardless of the style.

Reply:
I agree with you, so I went ahead and updated the test code to make use of this. 

I started down the path of adding a dependency on `delimiter` so that cases 1, 3, 4, and 5 in your example would use the given delimiter instead of `","` e.g. `a,,b`, `a--b`, `a//b`, and `a..b`. The code started getting messy, and needlessly complex, so I elected for a second simple test function to test that `SplitAndExtend` can handle different delimiters for a basic case.